controllers/rhmi/rhmi_controller.go

```diff
@@ -1367,6 +1384,108 @@ func (r *RHMIReconciler) createInstallationCR(ctx context.Context, serverClient
 	return installation, nil
 }
 
+func (r *RHMIReconciler) composeAndSetAlertMetric(installation *rhmiv1alpha1.RHMI, rc *rest.Config, configManager *config.Manager) error {
+	var alerts []resources.AlertMetric
+
+	alertingNamespaces, err := r.getAlertingNamespace(installation, configManager)
+	if err != nil {
+		return fmt.Errorf("getting alerting namespace failed: %w", err)
+	}
+
+	observability, err := configManager.ReadObservability()
+	if err != nil {
+		return fmt.Errorf("getting observability configuration failed: %w", err)
+	}
+
+	for namespace, _ := range alertingNamespaces {
+		if namespace == observability.GetNamespace() {
+			alerts, err = r.composeAlertMetric("prometheus", namespace, rc)
+			if err != nil {
+				return fmt.Errorf("composing alert metric failed: %w", err)
+			}
+			metrics.SetRHOAMAlerts(alerts)
+		}
+	}
+
+	return nil
+}
+
+func (r *RHMIReconciler) composeAlertMetric(route string, namespace string, rc *rest.Config) ([]resources.AlertMetric, error) {
+	var alerts resources.AlertMetrics
+	endpoint := "/api/v1/alerts"
+
+	url, err := r.getURLFromRoute(route, namespace, rc)
+	if err != nil {
+		return alerts.Alerts, fmt.Errorf("error getting route : %w", err)
+	}
+	url = url + endpoint
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return alerts.Alerts, fmt.Errorf("error on request : %w", err)
+	}
+	var bearer = "Bearer " + r.restConfig.BearerToken
+	req.Header.Add("Authorization", bearer)
+
+	client := &http.Client{}
+	client.Timeout = time.Second * 10
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return alerts.Alerts, fmt.Errorf("error on response : %w", err)
+	}
+
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			log.Error("reader close failure", err)
+		}
+	}(resp.Body)
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return alerts.Alerts, fmt.Errorf("unable to read body : %w", err)
+	}
+
+	var alertResp struct {
+		Status string `json:"status"`
+		Data   struct {
+			Alerts []struct {
+				Labels models.LabelSet `json:"labels"`
+				State  string          `json:"state"`
+			} `json:"alerts"`
+		} `json:"data"`
+	}
```

Comment:
```suggestion
	var alertResp struct {
		Status string `json:"status"`
		Data   struct {
				Alerts []v1.Alert "alerts"
		} `json:"data"`
	}
```

Where v1 is 
`v1 "github.com/prometheus/client_golang/api/prometheus/v1"`

Reply:
It took me a while too. 🙈